eth/beacon/state_machines/base.py

```diff
@@ -144,39 +149,44 @@ def active_state(self) -> ActiveState:
         Returns latest active state.
 
         It was backed up per cycle. The latest ActiveState could be reproduced by
-        ``old_active_state`` and recent blocks.
+        ``backup_active_state`` and recent blocks.
 
         NOTE: The following logic in beacon chain spec will be changed with the current spec.
         """
         if self._active_state is None:
             # Reproduce ActiveState
-            old_active_state_root = self.chaindb.get_active_state_root_by_crystallized(
+            backup_active_state_root = self.chaindb.get_active_state_root_by_crystallized(
                 self.crystallized_state.hash
             )
-            old_active_state = self.chaindb.get_active_state_by_root(old_active_state_root)
-            old_active_state_slot = self.crystallized_state.last_state_recalc
-
-            depth = self.config.CYCLE_LENGTH * 2
-            blocks = self.get_prev_blocks_to_slot(
-                self.parent_block.hash,
-                self.chaindb,
-                depth,
-                start_slot=old_active_state_slot
-            )
-            blocks = blocks[::-1]
-
-            self._active_state = self.get_active_state_class().from_old_active_and_blocks(
-                old_active_state,
-                blocks,
-                self.config.CYCLE_LENGTH * 2,
-            )
+            backup_active_state = self.chaindb.get_active_state_by_root(backup_active_state_root)
+            backup_active_state_slot = self.crystallized_state.last_state_recalc
+
+            if backup_active_state_root == self.parent_block.active_state_root:
+                # The backup ActiveState matches current block.
+                self._active_state = backup_active_state
+            else:
+                # Get recent blocks after last ActiveState backup.
+                searching_depth = self.config.CYCLE_LENGTH * 2
+                blocks = self.get_prev_blocks(
+                    last_block_hash=self.parent_block.hash,
+                    chaindb=self.chaindb,
+                    searching_depth=searching_depth,
+                    target_slot_number=backup_active_state_slot
+                )
+                blocks = blocks[::-1]
```

Comment:
I had to go check that this was reversing the `blocks`.  I think the idiomatic way to do this is to call `reversed()` on blocks, or better yet, to do `blocks = reversed(self.get_prev_blocks(...))` so that you don't have to define the `blocks` variable twice.